### PR TITLE
Add actionPresent Hooks

### PR DESCRIPTION
**Pull request by alexandrebrubeyonds**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.x
| Description?  | Add the 5 hooks adding in this PR https://github.com/PrestaShop/PrestaShop/pull/17203
| Fixed ticket? | No issue

### CODE REVIEW

**Comment by alexandrebrubeyonds** [comment]
Do I also have to do a PR for branch 8.x of docs ?

**Comment by PululuK** [comment]
> Do I also have to do a PR for branch 8.x of docs ?

ping @eternoendless

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
@alexandrebrubeyonds If this PR is approved and it's always synced with code with `develop`, then you will be able to create a PR for `8.x` for backporting.

File: modules/concepts/hooks/list-of-hooks.md [hunk 1]
```diff
@@ -1220,6 +1220,104 @@ actionGetAdminOrderButtons
...
+: 
+    Available since: {{< minver v="1.7.8" >}}
+
+    This hook is called before a cart is presented
+
+    Located in: /src/Adapter/Presenter/Cart/CartPresenter.php
+
+    Parameters:
+    ```php
+    <?php
```
Comment (eternoendless) [inline modules/concepts/hooks/list-of-hooks.md]: This is not needed anymore
```suggestion
```

File: modules/concepts/hooks/list-of-hooks.md [hunk 2]
```diff
@@ -1220,6 +1220,104 @@ actionGetAdminOrderButtons
...
+            'products_count' => $products_count,
+            'summary_string' => $summary_string,
+            'labels' => $labels,
+            'id_address_delivery' => $cart->id_address_delivery,
+            'id_address_invoice' => $cart->id_address_invoice,
+            'is_virtual' => $cart->isVirtualCart(),
+            'vouchers' => $vouchers,
+            'discounts' => $discounts,
+            'minimalPurchase' => $minimalPurchase,
+            'minimalPurchaseRequired' => Text minimal Purchase
```
Comment (eternoendless) [inline modules/concepts/hooks/list-of-hooks.md]: Perhaps these could be better described?

**Comment by alexandrebrubeyonds** [comment]
@eternoendless I provide more description for the params of hook `actionPresentCart`

File: modules/concepts/hooks/list-of-hooks.md (hunk 1 above)
Comment (kpodemski) [reply modules/concepts/hooks/list-of-hooks.md]: cc @alexandrebrubeyonds php tag is not needed anymore

File: modules/concepts/hooks/list-of-hooks.md [hunk 3]
```diff
@@ -1220,6 +1220,160 @@ actionGetAdminOrderButtons
...
+: 
+    Available since: {{< minver v="1.7.8" >}}
+
+    This hook is called before an order is presented
+
+    Located in: /src/Adapter/Presenter/Order/OrderPresenter.php
+
+    Parameters:
+    ```php
+    <?php
```
Comment (kpodemski) [inline modules/concepts/hooks/list-of-hooks.md]:
```suggestion
```

File: modules/concepts/hooks/list-of-hooks.md [hunk 4]
```diff
@@ -1220,6 +1220,160 @@ actionGetAdminOrderButtons
...
+: 
+    Available since: {{< minver v="1.7.8" >}}
+
+    This hook is called before an order return is presented
+
+    Located in: /src/Adapter/Presenter/Order/OrderReturnPresenter.php
+
+    Parameters:
+    ```php
+    <?php
```
Comment (kpodemski) [inline modules/concepts/hooks/list-of-hooks.md]:
```suggestion
```

File: modules/concepts/hooks/list-of-hooks.md [hunk 5]
```diff
@@ -1220,6 +1220,160 @@ actionGetAdminOrderButtons
...
+: 
+    Available since: {{< minver v="1.7.8" >}}
+
+    This hook is called before a product is presented
+
+    Located in: /src/Adapter/Presenter/Product/ProductPresenter.php
+
+    Parameters:
+    ```php
+    <?php
```
Comment (kpodemski) [inline modules/concepts/hooks/list-of-hooks.md]:
```suggestion
```

File: modules/concepts/hooks/list-of-hooks.md [hunk 6]
```diff
@@ -1220,6 +1220,160 @@ actionGetAdminOrderButtons
...
+: 
+    Available since: {{< minver v="1.7.8" >}}
+
+    This hook is called before a product listing is presented
+
+    Located in: /src/Adapter/Presenter/Product/ProductListingPresenter.php
+
+    Parameters:
+    ```php
+    <?php
```
Comment (kpodemski) [inline modules/concepts/hooks/list-of-hooks.md]:
```suggestion
```

File: modules/concepts/hooks/list-of-hooks.md [hunk 7]
```diff
@@ -1220,6 +1220,160 @@ actionGetAdminOrderButtons
...
+            'id_address_delivery' => (int) Delivery address ID,
+            'id_address_invoice' => (int) Invoice address ID,
+            'is_virtual' => (bool) If cart contains virtual product,
+            'vouchers' => array(
+                'allowed' => (int) If feature cart rule is active,
+                'added' => (array) List of vouchers,
+            ),
+            'discounts' => (array) List of highlighted discounts,
+            'minimalPurchase' => (string) Formatted price of configuration PS_PURCHASE_MINIMUM (and filter by hook overrideMinimalPurchasePrice),
+            'minimalPurchaseRequired' => Text minimal purchase if PS_PURCHASE_MINIMUM > Total products price without taxes
```
Comment (kpodemski) [inline modules/concepts/hooks/list-of-hooks.md:1303]: @alexandrebrubeyonds, I like the description of the array, but there are some issues with the indentation. Could you double-check it?

File: modules/concepts/hooks/list-of-hooks.md [hunk 8]
```diff
@@ -1220,6 +1220,160 @@ actionGetAdminOrderButtons
...
+            'summary_string' => (string) Number of products Text,
+            'labels' => array(
+                'tax_short' => Tax short label (incl or excl),
+                'tax_long' => Tax long label (include or exclude),
+            ),
+            'id_address_delivery' => (int) Delivery address ID,
+            'id_address_invoice' => (int) Invoice address ID,
+            'is_virtual' => (bool) If cart contains virtual product,
+            'vouchers' => array(
+                'allowed' => (int) If feature cart rule is active,
```
Comment (kpodemski) [inline modules/concepts/hooks/list-of-hooks.md]:
```suggestion
                'allowed' => (int) are CartRule's allowed?
```

File: modules/concepts/hooks/list-of-hooks.md [hunk 9]
```diff
@@ -1220,6 +1220,160 @@ actionGetAdminOrderButtons
...
+	    ),
+            'products_count' => (int) Number of products,
+            'summary_string' => (string) Number of products Text,
+            'labels' => array(
+                'tax_short' => Tax short label (incl or excl),
+                'tax_long' => Tax long label (include or exclude),
+            ),
+            'id_address_delivery' => (int) Delivery address ID,
+            'id_address_invoice' => (int) Invoice address ID,
+            'is_virtual' => (bool) If cart contains virtual product,
```
Comment (kpodemski) [inline modules/concepts/hooks/list-of-hooks.md]:
```suggestion
            'is_virtual' => (bool) true if cart contains at least one virtual product,
```

File: modules/concepts/hooks/list-of-hooks.md [hunk 10]
```diff
@@ -1220,6 +1220,160 @@ actionGetAdminOrderButtons
...
+		    'type' => 'tax'
+                    'label' => Price label
+                    'amount' => Price amount
+                    'value' => Price formatted
+		),
+	    ),
+            'products_count' => (int) Number of products,
+            'summary_string' => (string) Number of products Text,
+            'labels' => array(
+                'tax_short' => Tax short label (incl or excl),
```
Comment (kpodemski) [inline modules/concepts/hooks/list-of-hooks.md]:
```suggestion
                'tax_short' => Tax short label (incl. or excl.),
```

File: modules/concepts/hooks/list-of-hooks.md [hunk 11]
```diff
@@ -1220,6 +1220,160 @@ actionGetAdminOrderButtons
...
+                'tax_long' => Tax long label (include or exclude),
```
Comment (kpodemski) [inline modules/concepts/hooks/list-of-hooks.md]:
```suggestion
                'tax_long' => Tax long label (included or excluded),
```

isn't it with a `d` at the end?

File: modules/concepts/hooks/list-of-hooks.md [hunk 12]
```diff
@@ -1220,6 +1220,160 @@ actionGetAdminOrderButtons
...
+                    'value' => Price formatted
+		),
+		'tax' => array(
+		    'type' => 'tax'
+                    'label' => Price label
+                    'amount' => Price amount
+                    'value' => Price formatted
+		),
+	    ),
+            'products_count' => (int) Number of products,
```
Comment (kpodemski) [inline modules/concepts/hooks/list-of-hooks.md]:
```suggestion
            'products_count' => (int) Number of products in cart,
```

File: modules/concepts/hooks/list-of-hooks.md [hunk 13]
```diff
@@ -1220,6 +1220,160 @@ actionGetAdminOrderButtons
...
+            'summary_string' => (string) Number of products Text,
```
Comment (kpodemski) [inline modules/concepts/hooks/list-of-hooks.md]:
```suggestion
            'summary_string' => (string) Descriptive number of products,
```

**Comment by alexandrebrubeyonds** [comment]
@kpodemski I don't see issues with the indentation here : https://github.com/PrestaShop/docs/pull/1197#discussion_r736754273

You said that because there are too many characters on these lines ?

File: modules/concepts/hooks/list-of-hooks.md (hunk 7 above)
Comment (matks) [reply modules/concepts/hooks/list-of-hooks.md:1303]: I also cannot see indentation issues, like @alexandrebrubeyonds . @kpodemski can you have a look?

Else I think we can merge this contribution

**Review by matks — APPROVED** [review]

**Comment by matks** [comment]
Thank you @alexandrebrubeyonds

**Comment by matks** [comment]
@alexandrebrubeyonds could you submit a PR for 8.x branch 😉  ?

**Comment by alexandrebrubeyonds** [comment]
Yes, but sorry, I am very busy during this holiday season with our customers :)

**Comment by matks** [comment]
@alexandrebrubeyonds just do it when you're fine

happy to hear that, lot of work means lot of benefits 😄

File: modules/concepts/hooks/list-of-hooks.md (hunk 7 above)
Comment (matks) [reply modules/concepts/hooks/list-of-hooks.md:1303]: @kpodemski your review was right https://github.com/PrestaShop/docs/issues/1276

**Pull request merged**